hpx/lcos/promise.hpp

```diff
@@ -67,90 +67,90 @@ namespace lcos {
            > base_type;
 
     public:
-        // Effects: constructs a promise object and a shared state.
+        /// \brief constructs a promise object and a shared state.
         promise()
           : base_type()
         {}
 
-        // Effects: constructs a promise object and a shared state. The
-        // constructor uses the allocator a to allocate the memory for the
-        // shared state.
+        /// \breif constructs a promise object and a shared state. The
+        /// constructor uses the allocator a to allocate the memory for the
+        /// shared state.
         template <typename Allocator>
         promise(std::allocator_arg_t, Allocator const& a)
           : base_type(std::allocator_arg, a)
         {}
 
-        // Effects: constructs a new promise object and transfers ownership of
-        //          the shared state of other (if any) to the newly-
-        //          constructed object.
-        // Postcondition: other has no shared state.
+        /// \brief constructs a new promise object and transfers ownership of
+        ///          the shared state of other (if any) to the newly-
```

Comment:
Could you please check the indentation for multi-line blocks like this? It's a small thing but I think it would be nice to have it like so:
```
        /// \brief constructs a new promise object and transfers ownership of
        //         the shared state of other (if any) to the newly-
```
where `constructs` and `the` are aligned.